src/main/java/teammates/ui/webapi/action/SearchSessionsAction.java

```diff
@@ -0,0 +1,142 @@
+package teammates.ui.webapi.action;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+import teammates.common.util.StringHelper;
+import teammates.ui.webapi.output.SearchSessionData;
+import teammates.ui.webapi.output.SearchSessionsResult;
+
+/**
+ * Searches for sessions.
+ */
+public class SearchSessionsAction extends Action {
+
+    private static final String OPEN_CLOSE_DATES_SESSION_TEMPLATE = "[%s - %s]";
+
+    private Set<String> courseIds = new HashSet<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToOpenFeedbackSessionsMap = new HashMap<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToNotOpenFeedbackSessionsMap = new HashMap<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToPublishedFeedbackSessionsMap = new HashMap<>();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Only admins can search for sessions
+        if (!userInfo.isAdmin) {
+            throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String searchKey = getNonNullRequestParamValue(Const.ParamsNames.ADMIN_SEARCH_KEY);
+        List<StudentAttributes> students = logic.searchStudentsInWholeSystem(searchKey).studentList;
```

Comment:
In response to your question:

- From this line, you are not search for sessions but search for students, this call is still RPC oriented.
- I think you can safely remove the endpoint and move this join in the frontend.
    - Search for student to get a list of student.
    - Use `/sessions` endpoint to get sessions.

Reply:
We tried to implement the new endpoints using the existing logic as much as possible, that's why it's still the RPC style. The RESTful style would require a whole new set of logic (document, query, index, db) that would make the migration much larger, so we avoided that. 

Thanks for letting us know about `GetFeedbackSessionsAction`, we will use that and join on the frontend.